drivers/dma/dma_imx_rt.c

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright (c) 2020 NXP Semiconductor INC.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common part of DMA drivers for imx rt series.
+ */
+
+#include <errno.h>
+#include <soc.h>
+#include <init.h>
+#include <kernel.h>
+#include <drivers/dma.h>
+#include <drivers/clock_control.h>
+
+#include "dma_imx_rt.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dma_imx_rt, CONFIG_DMA_LOG_LEVEL);
+
+struct dma_imx_rt_config {
+	DMA_Type *base;
+	DMAMUX_Type *dmamux_base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+static __aligned(32) edma_tcd_t
+	tcdpool[DT_INST_0_NXP_IMX_EDMA_DMA_CHANNELS][CONFIG_DMA_TCD_QUEUE_SIZE];
+
+struct call_back {
+	edma_transfer_config_t transferConfig;
+	edma_handle_t edma_handle;
+	void *callback_arg;
+	void (*dma_callback)(void *callback_arg, u32_t channel, int error_code);
+	enum dma_channel_direction dir;
+	bool busy;
+};
+
+struct dma_imx_rt_data {
+	struct call_back data_cb[DT_INST_0_NXP_IMX_EDMA_DMA_CHANNELS];
+};
+
+#define DEV_CFG(dev)                                                           \
+	((const struct dma_imx_rt_config *const)dev->config->config_info)
+#define DEV_DATA(dev) ((struct dma_imx_rt_data *)dev->driver_data)
+#define DEV_BASE(dev) ((DMA_Type *)DEV_CFG(dev)->base)
+
+#define DEV_DMAMUX_BASE(dev) ((DMAMUX_Type *)DEV_CFG(dev)->dmamux_base)
+
+#define DEV_CHANNEL_DATA(dev, ch)                                              \
+	((struct call_back *)(&(DEV_DATA(dev)->data_cb[ch])))
+
+#define DEV_EDMA_HANDLE(dev, ch)                                               \
+	((edma_handle_t *)(&(DEV_CHANNEL_DATA(dev, ch)->edma_handle)))
+
+static void _edma_callback(edma_handle_t *handle, void *param,
```

Comment:
don't use leading underscores for driver functions

Reply:
ok